tests/acceptance/header-test.js

```diff
@@ -5,12 +5,32 @@ import { currentURL } from '@ember/test-helpers';
 import { module, test } from 'qunit';
 import { setupAnimationTest } from 'ember-animated/test-support';
 import { setupApplicationTest } from 'codecrafters-frontend/tests/helpers';
-import { signInAsSubscriber } from 'codecrafters-frontend/tests/support/authentication-helpers';
+import { signIn, signInAsSubscriber } from 'codecrafters-frontend/tests/support/authentication-helpers';
 
 module('Acceptance | header-test', function (hooks) {
   setupApplicationTest(hooks);
   setupAnimationTest(hooks);
 
+  test('header should show sign-in button if user is not authenticated', async function (assert) {
+    testScenario(this.server);
+
+    await catalogPage.visit();
+
+    assert.true(catalogPage.header.signInButton.isVisible, 'expect sign-in button to be visible');
+  });
+
+  test('header should show upgrade button if user does not have an active subscription', async function (assert) {
```

Comment:
Let's also add a test for header shows upgrade button if a user isn't signed in - this is important, there's no other way for a user to see our pricing unless this is visible, and we've heard complaints about this in the past